04-linear-chain.Rmd

```diff
@@ -0,0 +1,526 @@
+```{r, include = FALSE}
+ottrpal::set_knitr_image_path()
+```
+
+# Connecting multiple tasks together in a linear chain
+
+Now that you have a first task in a workflow up and running, the next step is to continue building out the workflow as described in our Workflow Plan:
+
+1.  `BwaMem` aligns the samples to the reference genome.
+2.  `MarkDuplicates` marks PCR duplicates.
+3.  `ApplyBaseRecalibrator` applies base recalibration.
+4.  `Mutect2` performs somatic mutation calling. For this current iteration, we start with `Mutect2TumorOnly` which only uses the tumor sample.
+5.  `annovar` annotates the called somatic mutations.
+
+We do this via a **linear chain**, in which we feed the output of one task to the input of the next task. Let's see how to build a linear chain in a workflow.
+
+## How to connect tasks together in a workflow
+
+We can easily connect tasks together because of the following: *The output variables of a task can be accessed at the workflow level as inputs for the subsequent task.*
+
+For instance, let's see the output of our `BwaMem` task.
+
+```         
+output {
+    File analysisReadyBam = "~{base_file_name}.aligned.bam"
+    File analysisReadySorted = "~{base_file_name}.sorted_query_aligned.bam"
+  }
+```
+
+The File variables `analysisReadyBam` and `analysisReadySorted` can now be accessed anywhere in the workflow block after the `BwaMem` task as `BwaMem.``analysisReadyBam` and `BwaMem.analysisReadySorted`, respectively. Then, we will take the output of `BwaMem` task into the input of the `MarkDuplicates` task: we call our next task `MarkDuplicates` with input `BwaMem.analysisReadySorted` from the `BwaMem` task:
```

Comment:
`BwaMem.``analysisReadyBam`  --> `BwaMem.analysisReadyBam` (may or may not affect rendering)

I'd also change the second sentence to be a bit more concise, eg "Therefore, when we call the`MarkDuplicates` task, we can pass it the input `BwaMem.analysisReadySorted` from the `BwaMem` task:"